packages/js/product-editor/src/blocks/schedule-sale/edit.tsx

```diff
@@ -66,10 +68,13 @@ export function Edit( {
 		}
 	}
 
-	// Hide and clean date fields if the user manually change
+	// Hide and clean date fields if the user manually changes
 	// the sale price to zero or less.
 	useEffect( () => {
-		if ( ! isSalePriceGreaterThanZero ) {
+		if (
+			edits.hasOwnProperty( 'sale_price' ) &&
```

Comment:
I think the following would be more readable...

```
'sale_price' in edits
```

Now, this is slightly different than using `hasOwnProperty`, as `in` will check the prototype chain. But, I personally think the readability gain makes it worth it. 

If not, maybe we could return a function such as `isFieldEdited` from `useProductEdits` that could be used as follows:

```
isFieldEdited( 'sale_price' )
```

Reply:
Or...

```
hasEdit( 'sale_price' )
```